lib/utils/device/ssh.ts

```diff
@@ -61,21 +61,20 @@ export async function performLocalDeviceSSH(
 		}
 
 		const serviceNames: string[] = [];
-		const containers = allContainers
-			.map((container) => {
-				for (const name of container.Names) {
-					if (regex.test(name)) {
-						return { id: container.Id, name };
-					}
+		const containers: Array<{ id: string; name: string }> = [];
```

Comment:
This change was to avoid `undefined` being part of the type of the `containers` array. Before this change, the type inferred by Typescript was: `Array<{ id: string; name: string } |  undefined>`.  Avoiding `undefined` allows avoiding the exclamation mark elsewhere, and simplified some other code I added.